answer.go

```diff
@@ -608,7 +608,7 @@ traversal:
 		if row := p.clients[cpath]; len(row) > 0 {
 			return row[0].client
 		}
-		c, pr := NewPromisedClient(pipelineClient{
+		c, pr := NewPromisedClient(PipelineClient{
 			p:         p,
```

Comment:
Are we intentionally keeping these fields private?  That is:  is it ever expected that users will instantiate a `PipelineClient` in their own code?

Reply:
I don't see a clear use case; we can always expose later. But I don't have a strong opinion.

Reply:
Yep, sounds good.  Just making sure it wasn't overlooked.  Resolving.